src/MongoDB.Driver.Core/Core/Operations/FindCommandOperation.cs

```diff
@@ -484,10 +484,11 @@ private AsyncCursor<TDocument> CreateCursor(IChannelSourceHandle channelSource,
         {
             var getMoreChannelSource = new ServerChannelSource(channelSource.Server, channelSource.Session.Fork());
             var firstBatch = CreateCursorBatch(commandResult);
+            var cursorDocument = commandResult["cursor"].AsBsonDocument;
```

Comment:
Consider refactoring 486-487 to:

    var cursorDocument = commandResult["cursor"].AsBsonDocument;
    var collectionNamespace = CollectionNamespace.FromFullName(cursorDocument["ns"].AsString);
    var firstBatch = CreateFirstCursorBatch(cursorDocument);

If you do this `CreateCursorBatch` will be renamed `CreateFirstCursorBatch` and its parameter will be a `cursorDocument` instead of a `commandResult`.

    private CursorBatch<TDocument> CreateFirstCursorBatch(BsonDocument cursorDocument)

Reply:
Done.